core/templates/pages/exploration-player-page/services/exploration-recommendations.service.ts

```diff
@@ -17,46 +17,106 @@
+import { Injectable } from '@angular/core';
+
+import { ContextService } from 'services/context.service';
+import { ServicesConstants } from 'services/services.constants';
+import { UrlService } from 'services/contextual/url.service';
+
+export interface ISummary {
+  category: string;
+  // eslint-disable-next-line camelcase
+  community_owned: boolean;
+  id: string;
+  // eslint-disable-next-line camelcase
+  language_code: string;
+  // eslint-disable-next-line camelcase
+  num_views: number;
+  objective: string;
+  status: string;
+  tags: string[];
+  // eslint-disable-next-line camelcase
+  thumbnail_bg_color: string;
+  // eslint-disable-next-line camelcase
+  thumbnail_icon_url: string;
+  title: string;
+}
+export interface ISummaries {
+  summaries: ISummary[];
+}
+type RecommendationsUrlParams = {
+  /* eslint-disable camelcase */
+  stringified_author_recommended_ids: string;
+  collection_id?: string;
+  story_id?: string;
+  current_node_id?: string;
+  include_system_recommendations: string;
+  /* eslint-enable camelcase */
+};
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ExplorationRecommendationsService {
+  constructor(
+    private contextService: ContextService,
+    private http: HttpClient,
+    private urlService: UrlService) {
+    ExplorationRecommendationsService.isIframed = this.urlService.isIframed();
+    ExplorationRecommendationsService.isInEditorPage = (
+      this.contextService.getPageContext() ===
+      ServicesConstants.PAGE_CONTEXT.EXPLORATION_EDITOR);
+    ExplorationRecommendationsService.isInEditorPreviewMode = (
+      ExplorationRecommendationsService.isInEditorPage && (
+        this.contextService.getEditorTabContext() ===
+        ServicesConstants.EXPLORATION_EDITOR_TAB_CONTEXT.PREVIEW));
+    ExplorationRecommendationsService.explorationId = (
+      this.contextService.getExplorationId());
+  }
+  static isIframed: boolean = null;
+  static isInEditorPage: boolean = null;
+  static isInEditorPreviewMode: boolean = null;
+  static explorationId: string = null;
+
+  getRecommendedSummaryDicts(
+      authorRecommendedExpIds: string[],
+      includeAutogeneratedRecommendations: boolean,
+      successCallback: (value?: Object | PromiseLike<Object>) => void): void {
+    var recommendationsUrlParams: RecommendationsUrlParams = {
+      stringified_author_recommended_ids: JSON.stringify(
+        authorRecommendedExpIds),
+      include_system_recommendations: 'false'
     };
+
+    var collectionId = this.urlService.getCollectionIdFromExplorationUrl();
+    var storyId = this.urlService.getUrlParams().story_id;
+    var currentNodeId = this.urlService.getUrlParams().node_id;
+    if (collectionId !== null) {
+      recommendationsUrlParams.collection_id = collectionId;
+    }
+    if (storyId !== null) {
+      recommendationsUrlParams.story_id = storyId;
+    }
+    if (currentNodeId !== null) {
+      recommendationsUrlParams.current_node_id = currentNodeId;
+    }
```

Comment:
This change is required since after upgrade the null params aren't automatically removed from url, so we need to manually check and add params.